src/callbacks.py

```diff
@@ -444,16 +333,16 @@ def compute_mean_distance(eval_dataset_questions, eval_dataset_defs, tag, mean_g
         # delete eval datasets and log metrics
         del eval_dataset_d1cons, eval_dataset_d2cons, eval_dataset_d1defs, eval_dataset_d2defs
 
-        self.tb_writer.add_scalar(f"eval/grad_mean_dist_{tag1}", mean_dist_d1, state.global_step)
-        self.tb_writer.add_scalar(f"eval/grad_mean_dist_{tag2}", mean_dist_d2, state.global_step)
+        self.tb_writer.add_tensor(f"eval/grad_mean_dist_{tag1}", torch.tensor(distances_d1), state.global_step)
+        self.tb_writer.add_tensor(f"eval/grad_mean_dist_{tag2}", torch.tensor(distances_d2), state.global_step)
         self.tb_writer.add_scalar("eval/grad_variance", variance, state.global_step)
         self.tb_writer.add_scalar("eval/grad_cosine_similarity", cos_sim, state.global_step)
-        self.tb_writer.add_scalar(f"eval/grad_mean_sim_{tag1}_cos", mean_sim_d1_cos, state.global_step)
-        self.tb_writer.add_scalar(f"eval/grad_mean_sim_{tag2}_cos", mean_sim_d2_cos, state.global_step)
+        self.tb_writer.add_tensor(f"eval/grad_mean_sim_{tag1}_cos", torch.tensor(sim_d1_cos), state.global_step)
+        self.tb_writer.add_tensor(f"eval/grad_mean_sim_{tag2}_cos", torch.tensor(sim_d2_cos), state.global_step)
         
         norms1.update(norms2)
         for norm in norms1:
-            self.tb_writer.add_scalar(f"eval/{norm}", norms1[norm], state.global_step)
+            self.tb_writer.add_tensor(f"eval/{norm}", torch.tensor(norms1[norm]), state.global_step)
```

Comment:
How can one load and plot this stuff (tensors in general)? Maybe add a little demo notebook showing this?

Reply:
I have such notebook I prepared for David earlier actually in collar